service/src/main/java/uk/gov/hmcts/reform/fpl/service/email/content/C2UploadedEmailContentProvider.java

```diff
@@ -24,14 +25,15 @@
     private final ObjectMapper mapper;
     private final Time time;
 
-    public C2UploadedTemplate buildC2UploadNotificationTemplate(final CaseDetails caseDetails) {
+    public C2UploadedTemplate buildC2UploadNotificationTemplate(final CaseDetails caseDetails,
+                                                                final DocumentReference latestC2) {
         CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
 
         C2UploadedTemplate adminTemplateForC2 = new C2UploadedTemplate();
         adminTemplateForC2.setCallout(buildCallout(caseData));
         adminTemplateForC2.setRespondentLastName(getFirstRespondentLastName(caseData.getRespondents1()));
         adminTemplateForC2.setCaseUrl(getCaseUrl(caseDetails.getId()));
-        adminTemplateForC2.setDocumentLink(linkToAttachedDocument(caseData.getSubmittedForm()));
+        adminTemplateForC2.setDocumentLink(linkToAttachedDocument(latestC2));
```

Comment:
looks like what we need is already in caseData.. would this work?
    adminTemplateForC2.setDocumentLink(linkToAttachedDocument(caseData.getLastC2DocumentBundle().getDocument()));

Reply:
Probably, but to be fair I prefer having the information "what was added" in the event, which makes it more decoupled. Probably wouldn't want to change this right now, but happy to refactor in the "restrict" bug if you insist :) 

@tomaszpowroznik what are your thoughts?

Reply:
agree, relying on the fact that something has been added as last is a bit slippery. In theory if we send notification  with @async, it could happened that something else has been added meanwhile to the case.

Reply:
Ok, I will leave as it is if you don't mind. Even if there is more code I believe my approach is easier to follow and more logically correct - we raise an event mentioning what was changed. But I am not much fussed and if you press I can refactor in the restrict "story" :)

Reply:
i agree what we have here is logically correct 👍